### PR TITLE
✨ Enable all golangci-lint plugins for virtual cluster project

### PR DESCRIPTION
**Pull request by m-messiah**
**What this PR does / why we need it**:

This is the last part from three-step golangci-lint enable, started in https://github.com/kubernetes-sigs/cluster-api-provider-nested/pull/285 and https://github.com/kubernetes-sigs/cluster-api-provider-nested/pull/286.

This part is focused on the gosimple, gosec, gocyclo and revive checkers, bringing little more readability and memory usage efficiency.

## Notes

1. The `revive` plugin has two exclusions, because otherwise it would require to write too much comments, which could not make a lot of sense:

```yaml
# EXC0012 revive: Annoying issue about not having a comment. The rare codebase has such comments
- exported (.+) should have comment( \(or a comment on this block\))? or be unexported
# EXC0013 revive: Annoying issue about not having a comment. The rare codebase has such comments
- package comment should be of the form "(.+)...
``` 

## Notable changes

1. kubectl-vc kubeconfig creation mode changed to 0600 instead of 0644. There is no sense to make config readable for other users, but if you have any objections - I could revert it and disable in `gosec` linter.
3. Minimum TLS version for TLSConfig is pinned to TLS1.2. I can't imaging anybody to use older and less secure, but it is still one of major changes
4. provisioner.ProvisionerAliyun is renamed to provisioner.Aliyun due to unnecessary duplication of the word _provisioner_
5. provisioner.ProvisionerNative is renamed to provisioner.Native due to unnecessary duplication of the word _provisioner_
6. validationplugin.ValidationPluginInterface is renamed to validationplugin.Interface due to unnecessary duplication of the word _validationplugin_
7. fairqueue.FifoQueue is exported type now

### CODE REVIEW

**Review by christopherhein — APPROVED** [review]
/lgtm

/cc @Fei-Guo do you have an opportunity to give this a scan as well? I can't see anything too concerning.

File: virtualcluster/pkg/syncer/syncer.go [hunk 1]
```diff
@@ -233,25 +233,6 @@ func (s *Syncer) Run(stopChan <-chan struct{}) {
 		}
 	}()
 	go wait.Until(s.healthPatrol, 1*time.Minute, stopChan)
-	go func() {
```
Comment (Fei-Guo) [inline virtualcluster/pkg/syncer/syncer.go:236]: @m-messiah  Why do you remove this go func?

Comment (m-messiah) [reply virtualcluster/pkg/syncer/syncer.go:236]: @Fei-Guo thanks for catching this up. I made a revert PR for this part of code: https://github.com/kubernetes-sigs/cluster-api-provider-nested/pull/288 

I retried it and it looks like when I hit `remove redundant return` in Goland - it removes this goroutine too..

Comment (christopherhein) [reply virtualcluster/pkg/syncer/syncer.go:236]: Thanks for catching that Fei!

**Pull request merged**